examples/seventh/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/beatlabs/patron"
+	"github.com/beatlabs/patron/examples"
+	"github.com/beatlabs/patron/log"
+	"github.com/beatlabs/patron/sync"
+	patronhttp "github.com/beatlabs/patron/sync/http"
+	tracehttp "github.com/beatlabs/patron/trace/http"
+	"github.com/beatlabs/patron/trace/kafka"
+	synckafka "github.com/beatlabs/patron/trace/kafka/sync"
+)
+
+const (
+	kafkaTopic  = "patron-topic"
+	kafkaBroker = "localhost:9092"
+)
+
+func init() {
+	err := os.Setenv("PATRON_LOG_LEVEL", "debug")
+	if err != nil {
+		fmt.Printf("failed to set log level env var: %v", err)
+		os.Exit(1)
+	}
+	err = os.Setenv("PATRON_JAEGER_SAMPLER_PARAM", "1.0")
+	if err != nil {
+		fmt.Printf("failed to set sampler env vars: %v", err)
+		os.Exit(1)
+	}
+
+	err = os.Setenv("PATRON_HTTP_DEFAULT_PORT", "50001")
+	if err != nil {
+		fmt.Printf("failed to set default patron port env vars: %v", err)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	name := "sixth"
+	version := "1.0.0"
+
+	err := patron.SetupLogging(name, version)
+	if err != nil {
+		fmt.Printf("failed to set up logging: %v", err)
+		os.Exit(1)
+	}
+
+	httpCmp, err := newHTTPComponent(kafkaBroker, kafkaTopic, "http://localhost:50000/sixth")
+	if err != nil {
+		log.Fatalf("failed to create processor %v", err)
+	}
+
+	// Set up routes
+	routes := []patronhttp.Route{
+		patronhttp.NewAuthGetRoute("/", httpCmp.sixth, true, nil),
+	}
+
+	srv := patron.New(name, version).WithRoutes(routes)
```

Comment:
You can actually call run after WithRoutes and no need to add a context variable just use it directly
```suggestion
	err := patron.New(name, version).WithRoutes(routes).Run(ctx)
```

Reply:
Sure; but I don't think it should be called `err`? Will check the code.